hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/AbstractScheduleHandler.java

```diff
@@ -29,17 +29,22 @@
 import com.hedera.node.app.spi.workflows.PreCheckException;
 import com.hedera.node.app.spi.workflows.PreHandleContext;
 import com.hedera.node.app.spi.workflows.PreHandleDispatcher;
+import java.util.Objects;
 
 /**
  * Provides some implementation support needed for both the {@link ScheduleCreateHandler} and {@link
  * ScheduleSignHandler}.
  */
 abstract class AbstractScheduleHandler {
+
+    private final PreHandleDispatcher dispatcher;
+
+    AbstractScheduleHandler(PreHandleDispatcher dispatcher) {
```

Comment:
Eck.  I wonder, would it work better to have the dispatcher available in the PreHandleContext, so we don't need to add it to constructors?  I don't know if the dispatchers are stateful or not, but making handlers less stateful seems a good goal.

Reply:
The dispatchers are going to be stateless "Singletons". (In quotes, because they will not be implemented as Singletons. It will be enforced by the DI.)

So far, the design has been, that only transaction-specific stuff goes into the `PreHandleContext`, while components that will be managed by DI should be injected through the constructor.